### PR TITLE
feat: add ability to use dynamic connection map

### PR DESCRIPTION
**Pull request by nulian**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -55,6 +55,32 @@ config :file_storage_api, :azure_blob,
...
+    access_key: "amazing"
+  }
+}
+```
+
+For S3
+```elixir
+%{
+  engine: :s3,
+  config: %{host: "test.docker", secret_key: "test123", access_key: "amazing", scheme: "http://"}
```
Comment (thomas9911) [inline README.md]: maybe you can do test.example instead of .docker

File: lib/api/azure/base.ex [hunk 2]
```diff
@@ -25,8 +25,22 @@ defmodule FileStorageApi.API.Azure.Base do
     end
   end
 
-  def container(container_name, connection_name) do
-    Container.new(storage(connection_name), container_name)
+  def storage(connection) do
+    config = Map.get(connection, :config, [])
+
+    if Map.get(config, :development) do
```
Comment (thomas9911) [inline lib/api/azure/base.ex]: the default is a list and you do Map.get here? Maybe you can use the Access module so it works for maps and keylists

File: lib/api/s3/base.ex [hunk 3]
```diff
@@ -59,4 +63,13 @@ defmodule FileStorageApi.API.S3.Base do
       option -> option
     end)
   end
+
+  defp http_opts do
+    storage_api = Application.get_env(:file_storage_api, :storage_api)
+
+    case Keyword.get(storage_api, :custom_ca_cert) do
```
Comment (thomas9911) [inline lib/api/s3/base.ex:72]: cant do Keyword.get on the nil default

Comment (nulian) [reply lib/api/s3/base.ex:72]: Added a list as fallback

File: README.md (hunk 1 above)
Comment (nulian) [reply README.md]: improved the example

File: lib/api/azure/base.ex (hunk 2 above)
Comment (nulian) [reply lib/api/azure/base.ex]: Changed all the Map.get from dynamic config to Access.get

**Review by thomas9911 — APPROVED** [review]

**Pull request merged**